src/pyuvsim/simsetup.py

```diff
@@ -1754,34 +1785,82 @@ def subselect(uv_obj, param_dict):
         "blt_inds",
     ]
 
-    # downselect baselines (or anything that can be passed to pyuvdata's select method)
-    # Note: polarization selection is allowed here, but will cause an error if the incorrect pols
-    # are passed to pyuvsim.
-    if "select" in param_dict:
-        select_params = param_dict["select"]
-        no_autos = bool(select_params.pop("no_autos", False))
-        select_params = {
-            k: v for k, v in select_params.items() if k in valid_select_keys
-        }
-        if "antenna_nums" in select_params:
-            select_params["antenna_nums"] = list(
-                map(int, select_params["antenna_nums"])
-            )
-        redundant_threshold = param_dict["select"].get("redundant_threshold", None)
-        if "bls" in select_params:
-            bls = select_params["bls"]
-            if isinstance(bls, str):
-                # If read from file, this should be a string.
-                bls = ast.literal_eval(bls)
-                select_params["bls"] = bls
-        if len(select_params) > 0:
-            uv_obj.select(**select_params)
+    select_params = param_dict["select"]
+    no_autos = bool(select_params.pop("no_autos", False))
+    select_params = {k: v for k, v in select_params.items() if k in valid_select_keys}
+    if "antenna_nums" in select_params:
+        select_params["antenna_nums"] = list(map(int, select_params["antenna_nums"]))
+    redundant_threshold = param_dict["select"].get("redundant_threshold", None)
+    if "bls" in select_params:
+        bls = select_params["bls"]
+        if isinstance(bls, str):
+            # If read from file, this should be a string.
+            bls = ast.literal_eval(bls)
+            select_params["bls"] = bls
+    if len(select_params) > 0:
+        uv_obj.select(**select_params)
+
+    if no_autos:
+        uv_obj.select(ant_str="cross")
+
+    if redundant_threshold is not None:
+        uv_obj.compress_by_redundancy(tol=redundant_threshold)
+
+
+def ordering(uv_obj, param_dict, reorder_blt_kw):
+    """Do conjugation/reordering on a UVData object.
+
+    Parameters
+    ----------
+    uv_obj : UVData
+        UVData object to do the selection on.
+    param_dict : dict
+        Dict of parameters that can include an `order` section.
+    reorder_blt_kw : dict
+        Dict giving order & minor order for blt reordering.
+
+    """
+    bl_conjugation_convention = None
+
+    # this is the order required by the simulator. Might as well default to it.
+    default_blt_order = ["time", "baseline"]
+    if "ordering" in param_dict:
+        ordering_dict = param_dict["ordering"]
+        if "blt_order" not in ordering_dict:
+            ordering_dict["blt_order"] = default_blt_order
+    else:
+        ordering_dict = {"blt_order": default_blt_order}
```

Comment:
You're correct, but I needed to set a default and this is what is actually enforced inside the simulator. Als, if you call `reorder_blts` and only specify time first, you'll get `["time", "baseline"]`, so it seemed a reasonable default.

Reply:
I suppose with the new `time_axis_faster_than_bls` and `blts_are_rectangular` attributes I can change the simulator to just check for those instead, I'll play with that.

Reply:
the time,baseline requirement was just to minimize re-compute on beams and coherency right?

Reply:
ah, if there's a good reason for it to be (time, baseline), then by all means. And I have no problem with having this as a default at all. I thought I saw that you were changing the ordering to (time, baseline) even if the user requested otherwise, and then simply changing the ordering back at the end. If so, that could be some time wasted if (time, baseline) is not strictly required and the user still requested something rectangular (like (time, ant1)).

Reply:
I actually think that the baseline order within a time does matter for computational efficiency because of the beam handling. So I'm not inclined to change what is done inside the simulator.

Reply:
OK, no problem